invisible_cities/cities/detsim_simulate_electrons.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+
+from typing import List
+from typing import Tuple
+
+# from invisible_cities.cities.detsim_get_psf import binedges_from_bincenters
+
+#######################################
+######### ELECTRON SIMULATION #########
+#######################################
+def generate_ionization_electrons(energies    : np.ndarray,
+                                  wi          : float = 22.,
+                                  fano_factor : float = 0.1) -> np.ndarray:
+    """ Generate secondary ionization electrons from energy deposits
+
+    Parameters:
+        :wi: float
+            Mean ionization energy
+        :fano_factor: float
+            Fano-factor. related with the deviation in ionization electrons
+        :energies: np.ndarray
+            Energy hits
+    Returns:
+        :nes: np.ndarray
+            The ionization electrons per hit
+
+    Comment:
+        The quotient energy/wi gives the mean value of the ionization electrons produced
+        in a hit (N).
+        The number ionization electrons produced in a hit (N) are assumed to be normally
+        distributed with mean N and variance (N*F). (Being F the fano_factor).
+        If the variance is such that <1, the electrons are assumed to be poisson distributed.
+    """
+    nes = energies / wi
+    var = nes * fano_factor
+    pois = var < 1      #See docstring for explanation
+    nes[ pois] = np.random.poisson(nes[pois])
+    nes[~pois] = np.round(np.random.normal(nes[~pois], np.sqrt(var[~pois])))
+    nes[nes<0] = 0
+    return nes.astype(int)
+
+
+def drift_electrons(zs             : np.ndarray,
+                    electrons      : np.ndarray,
+                    lifetime       : float = 12.,
+                    drift_velocity : float = 1.) -> np.ndarray:
+    """ Returns number of electrons due to lifetime losses from secondary electrons
+
+    Parameters:
+        :lifetime: float
+            Electron lifetime
+        :drift_velocity: float
+            Drift velocity at the active volume of the detector
+        :zs:
+            z coordinate of the ionization hits
+        :electrons:
+            Number of ionization electrons in each hit
+    Returns:
+        :nes: np.ndarray
+            Number of ionization electrons that reach the EL gap
+    """
+    @np.vectorize
+    def attachment(n_ie, t):
+        return np.count_nonzero(-lifetime * np.log(np.random.uniform(size=n_ie)) > t)
+
+    ts  = zs / drift_velocity
+    return attachment(electrons, ts)
+
+
+def diffuse_electrons(xs                     : np.ndarray,
+                      ys                     : np.ndarray,
+                      zs                     : np.ndarray,
+                      electrons              : np.ndarray,
+                      transverse_diffusion   : float = 1.,
+                      longitudinal_diffusion : float = 0.2)\
```

Comment:
```suggestion
                      transverse_diffusion   : float = 1 * units.mm / np.sqrt(units.cm),
                      longitudinal_diffusion : float = 0.2  * units.mm / np.sqrt(units.cm))\
```
Again, the most recent numbers are 1.1 and 0.3, I think.

Reply:
I was using 1 and 0.2 in the last productions

Reply:
@paolafer Which values should we set as default?

Reply:
The current values of NEXT100, extrapolated from the measurements made in NEW, are 1 and 0.2. I don't have a strong opinion on which values the default should have: these functions are general, and not meant to simulate NEXT100 in this conditions of temperature and pressure only.